processor/src/advice/mem_provider.rs

```diff
@@ -12,72 +12,73 @@ use super::{
 #[derive(Debug, Clone, Default)]
 pub struct MemAdviceProvider {
     step: u32,
-    tape: Vec<Felt>,
-    values: BTreeMap<[u8; 32], Vec<Felt>>,
+    stack: Vec<Felt>,
+    map: BTreeMap<[u8; 32], Vec<Felt>>,
     sets: BTreeMap<[u8; 32], MerkleSet>,
 }
 
 impl From<AdviceInputs> for MemAdviceProvider {
     fn from(inputs: AdviceInputs) -> Self {
-        let (mut tape, values, sets) = inputs.into_parts();
-        tape.reverse();
+        let (mut stack, map, sets) = inputs.into_parts();
+        stack.reverse();
         Self {
             step: 0,
-            tape,
-            values,
+            stack,
+            map,
             sets,
         }
     }
 }
 
 impl AdviceProvider for MemAdviceProvider {
-    // ADVICE TAPE
+    // ADVICE STACK
     // --------------------------------------------------------------------------------------------
 
-    fn read_tape(&mut self) -> Result<Felt, ExecutionError> {
-        self.tape.pop().ok_or(ExecutionError::AdviceTapeReadFailed(self.step))
+    fn pop_stack(&mut self) -> Result<Felt, ExecutionError> {
+        self.stack.pop().ok_or(ExecutionError::AdviceStackReadFailed(self.step))
     }
 
-    fn read_tape_w(&mut self) -> Result<Word, ExecutionError> {
-        if self.tape.len() < 4 {
-            return Err(ExecutionError::AdviceTapeReadFailed(self.step));
+    fn pop_stack_word(&mut self) -> Result<Word, ExecutionError> {
+        if self.stack.len() < 4 {
+            return Err(ExecutionError::AdviceStackReadFailed(self.step));
         }
 
-        let idx = self.tape.len() - 4;
-        let result = [self.tape[idx + 3], self.tape[idx + 2], self.tape[idx + 1], self.tape[idx]];
+        let idx = self.stack.len() - 4;
+        let result =
+            [self.stack[idx + 3], self.stack[idx + 2], self.stack[idx + 1], self.stack[idx]];
 
-        self.tape.truncate(idx);
+        self.stack.truncate(idx);
 
         Ok(result)
     }
 
-    fn read_tape_dw(&mut self) -> Result<[Word; 2], ExecutionError> {
-        let word0 = self.read_tape_w()?;
-        let word1 = self.read_tape_w()?;
+    fn pop_stack_dword(&mut self) -> Result<[Word; 2], ExecutionError> {
+        let word0 = self.pop_stack_word()?;
+        let word1 = self.pop_stack_word()?;
 
         Ok([word0, word1])
     }
 
-    fn write_tape(&mut self, source: AdviceSource) -> Result<(), ExecutionError> {
+    fn write_stack(&mut self, source: AdviceSource) -> Result<(), ExecutionError> {
```

Comment:
Should we rename this to `push_stack()`?

Reply:
That makes sense; done!